public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -106,6 +107,47 @@ async function fetchDashboard(
   }
 }
 
+const queriesOnInitDashboard = (
+  panels: PanelModel[],
+  dashboardId: string,
+  orgId?: number,
+  userId?: number,
+  grafanaVersion?: string
+): void => {
+  let queries: { [key: string]: DataQuery[] } = getQueriesFromDashboardPerDS(panels, {});
+  const datasourceSrv = getDataSourceSrv();
+
+  Object.keys(queries).forEach((key) => {
+    datasourceSrv.get(key).then((value) => {
+      if (value.queriesOnInitDashboard) {
+        value.queriesOnInitDashboard(queries[key], dashboardId, orgId, userId, grafanaVersion);
+      }
+    });
+  });
+};
+
+const getQueriesFromDashboardPerDS = (
+  panels: PanelModel[],
+  queries: { [key: string]: DataQuery[] } = {}
+): { [key: string]: DataQuery[] } => {
+  panels.forEach((panel) => {
```

Comment:
```suggestion
  queries?: { [key: string]: DataQuery[] },
): { [key: string]: DataQuery[] } => {
  if(!queries) {
    queries = {}
  }
  panels.forEach((panel) => {
```

so you can call it just like `getQueriesFromDashboardPerDS(panels)`

Reply:
Actually that's exactly what the default parameter ` = {}` does on line 131.
But you're right, I don't need to add an empty object for queries when I call it first, I'll remove it 🚀